block/src/Pos/Block/Network/Retrieval.hs

```diff
@@ -308,3 +309,50 @@ getProcessBlocks pm diffusion nodeId desired checkpoints = do
                   else pure False
           when exitedRecovery $
               logInfo "Recovery mode exited gracefully on receiving block we needed"
+
+streamProcessBlocks
+    :: forall ctx m.
+       ( BlockWorkMode ctx m
+       , HasMisbehaviorMetrics ctx
+       )
+    => ProtocolMagic
+    -> Diffusion m
+    -> NodeId
+    -> HeaderHash
+    -> [HeaderHash]
+    -> m ()
+streamProcessBlocks pm diffusion nodeId desired checkpoints = do
+    logInfo "streaming start"
+    r <- Diffusion.streamBlocks diffusion nodeId desired checkpoints (loop (0::Word32) [])
+    case r of
+         Nothing -> do
+             logInfo "streaming not supported, reverting to batch mode"
+             getProcessBlocks pm diffusion nodeId desired checkpoints
+         Just _  -> do
+             logInfo "streaming done"
+             return ()
+  where
+    loop n blocks (streamWindow, wqgM, blockChan) = do
```

Comment:
What's the significance of the name `wqgM`?

Reply:
Also, should force `blocks`.

Reply:
wqgM = Write Queue Gauge Maybe.

Reply:
Will do `!blocks`.